noir-projects/aztec-nr/aztec/src/messages/encryption/aes128.nr

```diff
@@ -34,9 +34,54 @@ use crate::{
 
 use std::aes128::aes128_encrypt;
 
-fn extract_close_to_uniformly_random_256_bits_from_ecdh_shared_secret_using_poseidon2(
+/**
+ * Unsafe because of https://eprint.iacr.org/2010/264.pdf Page 13, Lemma 2 (and the 
```

Comment:
I'd make the docstring above the fn be just 'this is what this fn does, it's unsafe unless you do X', and then have the detailed explanation inside.